### ISSUE TITLE
implement iterator API

### ISSUE THREAD

**Issue by burgerdev** [burgerdev/ocaml-unqlite#9]
Maybe stick to existing `Map` signatures.

**Comment by burgerdev** [comment]
`Map` is a pure interface and thus does not play well with the side-effecting unqlite bindings. E.g.:

```ocaml
module type Map.S = sig
  (* how to do this? *)
  val empty : 'a t
end
```

**Comment by burgerdev** [comment]
Iterator was implemented in fd71c7c50119005f09c6d3872303ae968b437763
